Library/Homebrew/extend/ARGV.rb

```diff
@@ -13,7 +13,7 @@ def flags_only
 
   def formulae
     require "formula"
-    @formulae ||= (downcased_unique_named - casks).map { |name| Formulary.factory(name, spec) }
+    @formulae ||= (downcased_unique_named - casks).map { |name| Formulary.factory_with_priority(name, spec) }
```

Comment:
Let's not alter `Formulary#factory`. Instead, do it in the way similar to `ARGV#resolved_formulae`:

```ruby
   def formulae
     require "formula"
    @formulae ||= (downcased_unique_named - casks).map do |name|
       if name.include?("/")
         Formulary.factory(name, spec)
       else
         Formulary.find_with_priority(name, spec)
       end
     end
```

That's `Formulary#find_with_priority` only accept formula name(not url, not full-qualified name). And it should be something like:
```ruby
def find_with_priority(name, spec)
  possible_pinned_tap_formulae = pinned_tap_paths(ref)
  if possible_pinned_tap_formulae.size > 1
    raise TapFormulaAmbiguityError.new(ref, possible_pinned_tap_formulae)
  elsif possible_pinned_tap_formulae.size == 1
    selected_formula = possible_pinned_tap_formulae.first
    if core_path(ref).file?
      opoo "#{ref} is provided by core, but is now shadowed by #{selected_formula.tap_name}."
     end
    factory(selected_formula, spec)
  else
    factory(name, spec)
  end
end
```